interactivity/configuring-ai-powered-insights.md

```diff
@@ -0,0 +1,428 @@
+{
+    if (reportSource.Report == "report-suitable-for-markdown-output.trdp")
+    {
+        aiThreadInfo.PredefinedPrompts.Add("Create a summary of the report in Markdown (.md) format.");
+    }
+
+    base.UpdateAIPrompts(reportSource, aiThreadInfo);
+}
+````
+
+
+### GetAIResponse(string, string, string, string, AIQueryArgs)
+
+The [GetAIResponse(string, string, string, string, AIQueryArgs)](/api/telerik.reporting.services.webapi.reportscontrollerbase#Telerik_Reporting_Services_WebApi_ReportsControllerBase_GetAIResponse_System_String_System_String_System_String_System_String_Telerik_Reporting_Services_Engine_AIQueryArgs_) method is called every time a prompt is sent to the AI model. This method provides control over the AI request workflow, allowing you to intercept, modify, and validate requests before they reach the LLM. Below are examples of common customization scenarios.
+
+#### .NET
+
+````Modifying·Outgoing·Prompts
+/// <summary>
+/// Modifies the prompt sent from the client before passing it to the LLM.
+/// </summary>
+/// <returns></returns>
+public override async Task<IActionResult> GetAIResponse(string clientID, string instanceID, string documentID, string threadID, AIQueryArgs args)
+{
+    args.Query += $"{Environment.NewLine}Keep your response concise.";
+
+    return await base.GetAIResponse(clientID, instanceID, documentID, threadID, args);
+}
+````
+````Token·Usage·Validation
+/// <summary>
+/// Examines the approximate tokens count and determines whether the prompt should be sent to the LLM.
+/// </summary>
+/// <returns></returns>
+public override async Task<IActionResult> GetAIResponse(string clientID, string instanceID, string documentID, string threadID, AIQueryArgs args)
+{
+    const int MAX_TOKEN_COUNT = 500;
+    args.ConfirmationCallBack = (AIRequestInfo info) =>
+    {
+        if (info.EstimatedTokensCount > MAX_TOKEN_COUNT)
+        {
+            return ConfirmationResult.CancelResult($"The estimated token count exceeds the allowed limit of {MAX_TOKEN_COUNT} tokens.");
+        }
+
+        return ConfirmationResult.ContinueResult();
+    };
+
+    return await base.GetAIResponse(clientID, instanceID, documentID, threadID, args);
+}
+````
+````RAG·Optimization·Monitoring
+/// <summary>
+/// Examines whether the RAG optimization is applied for the current prompt.
+/// </summary>
+/// <returns></returns>
+public override async Task<IActionResult> GetAIResponse(string clientID, string instanceID, string documentID, string threadID, AIQueryArgs args)
+{
+    args.ConfirmationCallBack = (AIRequestInfo info) =>
+    {
+        if (info.Origin == AIRequestInfo.AIRequestOrigin.Client)
+        {
+            System.Diagnostics.Trace.TraceInformation($"RAG optimization is {info.RAGOptimization} for this prompt.");
+        }
+
+        return ConfirmationResult.ContinueResult();
+    };
+
+    return await base.GetAIResponse(clientID, instanceID, documentID, threadID, args);
+}
+````
+
+#### .NET Framework
```

Comment:
I know it is already added as information, but maybe reiterate in that section as a note that the RAG-specific example is missing from the .NET Framework examples because RAG functionality is .NET and .NET Standard only.